apps/client/ui/SortingSelect/SortingSelect.tsx

```diff
@@ -0,0 +1,140 @@
+  const ulRef = useRef<HTMLUListElement>(null);
+  const liRefs = useRef<(HTMLLIElement | null)[]>([]);
+
+  function selectOption(option: string) {
+    dispatch(selectSortOption(option));
+    setIsOpen(false);
+  }
+
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (
+        selectRef.current &&
+        !selectRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    }
+    document.addEventListener('click', handleClickOutside);
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+    };
+  }, []);
+
+  useEffect(() => {
+    const liHeight = liRefs.current[focusedIndex]?.offsetHeight || 0;
+    const scrollTop = ulRef.current?.scrollTop || 0;
+    const viewport = scrollTop + (ulRef.current?.offsetHeight || 0);
+    const liOffset = liHeight * focusedIndex;
+
+    if (
+      ulRef.current &&
+      (liOffset < scrollTop || liOffset + liHeight > viewport)
+    ) {
+      ulRef.current.scrollTop = liOffset;
+    }
+  }, [focusedIndex]);
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLButtonElement>) => {
+    if (!isOpen) return;
+    event.preventDefault();
+    switch (event.key) {
+      case 'ArrowUp':
+        setFocusedIndex((prevIndex) => (prevIndex <= 0 ? 0 : prevIndex - 1));
+        break;
+      case 'ArrowDown':
+        setFocusedIndex((prevIndex) =>
+          prevIndex >= options.length - 1 ? options.length - 1 : prevIndex + 1,
+        );
+        break;
+      case 'Enter':
+      case ' ':
+        if (focusedIndex >= 0 && focusedIndex < options.length) {
+          selectOption(options[focusedIndex]);
+        }
+        break;
+      case 'Escape':
+        setIsOpen(false);
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <div
+      className={clsx('relative border-r border-dark-grey', className)}
+      ref={selectRef}
+    >
+      <button
+        className={clsx(
+          'flex items-center justify-between bg-background-color',
+          'focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-gray-300',
+        )}
+        onClick={() => setIsOpen(!isOpen)}
+        onKeyDown={handleKeyDown}
+        type="button"
+      >
+        <span className="pr-1.5 text-sm font-bold">{selected}</span>
+        <ArrowDownIcon
+          className={clsx(
+            'transform transition-transform duration-100 ease-in-out',
+            { 'rotate-180': isOpen },
+          )}
+        />
+      </button>
+      {isOpen && (
+        <ul
+          className="absolute right-0 top-full z-10 mt-1 w-48 rounded-md border bg-white"
+          ref={ulRef}
+        >
+          {options.map((option, index) => (
+            <li
+              role="menuitem"
+              key={option}
+              className={clsx(
+                'cursor-pointer px-4 py-2 text-xs text-black',
+                'transition-ease-in-out p-3 transition-colors duration-200 hover:bg-gray-300',
```

Comment:
the `hover:` here is the reason of 2 focused elements when you use keyboard and then mouse 👌 the hover state works separately and it is not related with the state set with keyboard - to resolve the issue try to have one source of true for focus/hover state 😈

Reply:
ok, that was tricky but works now 😎